error.go

```diff
@@ -20,295 +20,233 @@
 package goa
 
 import (
-	"bytes"
-	"encoding/json"
 	"fmt"
-	"strconv"
 	"strings"
 )
 
-type (
-	// ErrorID is an enum listing the possible types of errors.
-	ErrorID int
-
-	// TypedError describes an error that can be returned in a HTTP response.
-	TypedError struct {
-		ID   ErrorID
-		Mesg string
-	}
-
-	// MultiError records multiple errors.
-	MultiError []error
-
-	// BadRequestError is the type of errors that result in a response with status code 400.
-	BadRequestError struct {
-		Actual error
-	}
+const (
+	errInvalidParamTypeID     = "goa.ipt"
+	errMissingParamID         = "goa.mpa"
+	errInvalidAttributeTypeID = "goa.iat"
+	errMissingAttributeID     = "goa.mat"
+	errInvalidEnumValueID     = "goa.iev"
+	errMissingHeaderID        = "goa.mhe"
+	errInvalidFormatID        = "goa.ifo"
+	errInvalidPatternID       = "goa.ipa"
+	errInvalidRangeID         = "goa.ira"
+	errInvalidLengthID        = "goa.ile"
+	errInvalidEncodingID      = "goa.ien"
+	errInternalID             = "goa.int"
```

Comment:
why not get rid of those `goa.ipt` IDs, convert the strings we have `missing required parameter` to `missing_required_parameter`, making it clear it is an identifier rather than a freeform message.

We will have a freeform error description through `Errorf()` or `Error()` .. let that be the "title" or "description" or "error_message". What about `error` instead of `id` ? is it always clear that it is an error object that we have at hand ?

I'd imagine:

```
{"error": "missing_parameter_thing",
 "error_message": "The parameter 'superbob' was missing from 'whatever.struct'",
 "status": 400}
```

and have the status reflect that.. (or no `status` prop. in there).

However a user wants to display that error with some custom error handlers, there's no confusion as to the nature of that object :)